app/models/account/search.rb

```diff
@@ -60,7 +60,9 @@ def search_ids(query, options = {})
                                  per_page: ThreeScale::Search::Helpers::SPHINX_PAGE_SIZE_INFINITE,
                                  ignore_scopes: true, with: { })
 
-        options.deep_merge!(with: { provider_account_id: User.current.account_id })
+        if (account_id = User.account_id)
```

Comment:
I much more prefer `User.current.account_id`, I was looking at this for a minute and was wondering what this should mean.

Reply:
On the other hand I see that originally it was `User.tenant_id` which I also don't like. But at least it seems consistent.

Reply:
The thing though is that this `provider_account_id: account_id` will only work for accounts. Other objects don't have this property but have the `tenant_id` property. So I'm concerned search will not properly work for other object types.

I think this condition is great in case we are searching for accounts. Maybe add the `provider_account_id: account_id` only when searching accounts.

Reply:
> Maybe add the provider_account_id: account_id only when searching accounts.

But this code is under `Account::Search`, is it expected to be used to search for other resources?

> I much more prefer `User.current.account_id`

Yeah you're right. `User.account_id` is only there to not have to worry about `User.current` being `nil`, but the name is confusing. I'll fix it.